lib/plottr_components/src/components/outline/BeatView.js

```diff
@@ -158,6 +166,30 @@ const BeatViewConnector = (connector) => {
       )
     }
 
+    componentDidMount() {
+      document.addEventListener('click', this.handleClickOutside, true)
+    }
+
+    componentWillUnmount() {
+      document.removeEventListener('click', this.handleClickOutside, true)
+    }
+
+    handleClickOutside = (e) => {
+      if (this.titleRef.current && this.titleRef.current.contains(e.target)) {
```

Comment:
I'm afraid I have to disagree with the timeout here because it adds extra complexity and re-renders.
What if we initialised a piece of state in the `CardView` to a timestamp.
Then you can set `clickedOutside` to a timestamp in `BeatView` whenever the user clicks outside.
The `CardView` can compare the timestamp to its own.
If the timestamp is different, it calls `saveEdit`.

One other problem here is that the if statements should be combined because the body of the first if statement is identical to the second.

Reply:
@Quiescent I removed the setTimeout and it didn't break anything, but im not completely sure if that has another downside though. What do you think Ed?

Reply:
Ah!  The callback argument of `setState`.
Yes.  That'll do.

Reply:
I see that you didn't collapse the if statements.
Because the whole block is so small, I'm not going to block merging because of it, but I'm going to point out that the body of the first condition and the else are the same.